File: web_testing/watir/pages/bbc_homepage.rb
class Bbc_homepage

    def initialize(browser)
        @browser = browser

        #page objects
        @homepage_url = 'https://www.bbc.co.uk/'
        @sign_in_link_id = "idcta-link"
    end

    def access_homepage
        @browser.goto(@homepage_url)
    end

    def sign_in_link
        @browser.link(id: @sign_in_link_id)
    end

    def click_sign_in_link
        sign_in_link.click
    end

end

File: BbcTest/BbcTest/pages/BbcHomepage.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace BbcTest.pages
{
    public class BbcHomepage
    {
        private IWebDriver driver;

        public BbcHomepage(IWebDriver driver)
        {
            this.driver = driver;
            PageFactory.InitElements(driver, this);
        }

        [FindsBy(How = How.Id, Using = "idcta-link")]
        private IWebElement SignInLink;

        public void GoToPage()
        {
            driver.Navigate().GoToUrl("https://bbc.co.uk");
        }

        public void ClickSignInLink()
        {
            SignInLink.Click();
        }

    }
}
